math/ncr_modulo_p.cpp

```diff
@@ -0,0 +1,122 @@
+/**
+ * @file
+ * @brief This program aims at calculating nCr modulo p
+ *
+ */
+
+#include <cassert>
+#include <iostream>
+#include <vector>
+
+/** Class which contains all methods required for calculating nCr mod p
+ *
+ */
+class NCRModuloP {
+ private:
+    std::vector<int64_t> fac;
```

Comment:
Please use `uint64_t` instead (same in other parts of the code).

Reply:
@Panquesito7 I have replaced `int64_t` with `uint64_t` in all parts except the `ncr` and `modInverse` function because those functions can return a negative value in case the modular inverse doesn't exist. Also, the variables x,y in `gcdExtended` should be `int64_t` because they can be negative as well.